src/NzbDrone.Core/Download/Clients/QBittorrent/QBittorrentStalledPolicy.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+
+namespace NzbDrone.Core.Download.Clients.QBittorrent;
+
+public interface IQBittorrentStalledPolicy
+{
+    void Apply(QBittorrentTorrent torrent, DownloadClientItem ditem, QBittorrentSettings settings);
+}
+
+public interface IDateTimeProvider
+{
+    DateTime Now { get; }
+}
+
+internal sealed class DateTimeProvider : IDateTimeProvider
+{
+    public DateTime Now => DateTime.Now;
+}
+
+public sealed class QBittorrentStalledPolicy : IQBittorrentStalledPolicy
+{
+    private readonly IDateTimeProvider _dateTime;
+
+    public QBittorrentStalledPolicy(IDateTimeProvider dateTime)
+    {
+        _dateTime = dateTime;
+    }
+
+    public void Apply(QBittorrentTorrent torrent, DownloadClientItem ditem, QBittorrentSettings settings)
+    {
+        if (torrent.State != "stalledDL")
+        {
+            return;
+        }
+
+        if (this.ApplyPolicy(torrent, settings))
```

Comment:
```suggestion
        if (ApplyPolicy(torrent, settings))
```

Though I think calling out to another function is unneccesarry here as `HasNotBeenActive` and `HasBeenInQueueLongEnough` are already their own functions.